src/Collapse.js

```diff
@@ -1,234 +1,169 @@
 import React from 'react';
-import {shouldComponentUpdate} from 'react/lib/ReactComponentWithPureRenderMixin';
 import {Motion, spring} from 'react-motion';
-import HeightReporter from 'react-height';
+import {shouldComponentUpdate} from 'react/lib/ReactComponentWithPureRenderMixin';
+
 
+const SPRING_PRECISION = 1;
 
-const PRECISION = 0.5;
 
+const WAIT = 'WAIT';
+const RESIZE = 'RESIZE';
+const REST = 'REST';
+const STABLE = 'STABLE';
 
-const stringHeight = height => Math.max(0, parseFloat(height)).toFixed(1);
 
+const noop = () => null;
+const css = {
+  collapse: 'collapse',
+  content: 'content'
+};
 
-const Collapse = React.createClass({
+
+export const Collapse = React.createClass({
   propTypes: {
     isOpened: React.PropTypes.bool.isRequired,
-    children: React.PropTypes.node.isRequired,
-    fixedHeight: React.PropTypes.number,
-    style: React.PropTypes.object, // eslint-disable-line react/forbid-prop-types
     springConfig: React.PropTypes.objectOf(React.PropTypes.number),
-    keepCollapsedContent: React.PropTypes.bool,
+
+    theme: React.PropTypes.objectOf(React.PropTypes.string),
+    style: React.PropTypes.object,
+
     onRest: React.PropTypes.func,
-    onHeightReady: React.PropTypes.func
+    onHeightReady: React.PropTypes.func,
+
+    children: React.PropTypes.node.isRequired
   },
 
 
   getDefaultProps() {
     return {
-      fixedHeight: -1,
       style: {},
-      keepCollapsedContent: false,
-      onHeightReady: () => {} // eslint-disable-line no-empty-function
+      theme: css,
+      onRest: noop,
+      onHeightReady: noop
     };
   },
 
 
   getInitialState() {
-    return {height: -1, isOpenedChanged: false};
+    return {action: STABLE, from: 0, to: 0};
   },
 
-  componentWillMount() {
-    this.height = stringHeight(0);
-    this.renderStatic = true;
+
+  componentDidMount() {
+    if (this.props.isOpened) {
+      const to = this.content.clientHeight;
+      this.setState({action: STABLE, from: to, to});
+    }
+    this.props.onRest();
```

Comment:
Because the callbacks are not methods of `this.props` object. I generally avoid calling such functions as methods because this changes semantics of what is expected inside these functions (I expect strictly no `this` inside).